database/db-driver/greenplum/db-assign_sequential_id

```diff
@@ -1,97 +1,93 @@
+                      FROM ' || QUOTE_IDENT(tname) || '
+                     GROUP BY gpsid
+                     ORDER BY gpsid
+                 ) histogram;';
+      DECLARE
+        gpsids          INT[]    := ARRAY(SELECT gpsid          FROM dd_tmp_id_ranges_by_gpsid ORDER BY gpsid);
+        cumulative_cnts BIGINT[] := ARRAY(SELECT cumulative_cnt FROM dd_tmp_id_ranges_by_gpsid ORDER BY gpsid);
+        cnts            BIGINT[] := ARRAY(SELECT cnt            FROM dd_tmp_id_ranges_by_gpsid ORDER BY gpsid);
+      BEGIN
+        EXECUTE 'SELECT fast_seqassign_init();';
+        EXECUTE 'UPDATE ' || tname || ' SET ' || cname || ' = fast_seqassign_next(
+            ' || startid || ',
+            ' || increment || ',
+            gp_segment_id,
+            ARRAY[' || ARRAY_TO_STRING(         gpsids, ',')::TEXT || '],
+            ARRAY[' || ARRAY_TO_STRING(cumulative_cnts, ',')::TEXT || '],
+            ARRAY[' || ARRAY_TO_STRING(           cnts, ',')::TEXT || ']
+          );';
+        RETURN 'fast_seqassign done for segments [' || ARRAY_TO_STRING(gpsids, ',') || ']' ||
+               ' with counts [' || ARRAY_TO_STRING(cnts, ',') || ']';
+      END;
     END;
     \$\$ LANGUAGE 'plpgsql';
 
-    CREATE OR REPLACE FUNCTION _fast_seqassign(tname character varying, startid bigint)
-    RETURNS TEXT AS
-    \$\$
-    DECLARE
-      sids int[] :=  ARRAY(SELECT sid FROM tmp_gpsid_count ORDER BY sid);
-      base_ids bigint[] :=  ARRAY(SELECT base_id FROM tmp_gpsid_count ORDER BY sid);
-      base_ids_noagg bigint[] :=  ARRAY(SELECT base_id FROM tmp_gpsid_count_noagg ORDER BY sid);
-      tsids text;
-      tbase_ids text;
-      tbase_ids_noagg text;
-    BEGIN
-      SELECT INTO tsids array_to_string(sids, ',');
-      SELECT INTO tbase_ids array_to_string(base_ids, ',');
-      SELECT INTO tbase_ids_noagg array_to_string(base_ids_noagg, ',');
-      if ('update ' || tname || ' set ' || cname || ' = updateid(' || startid || ', gp_segment_id, ARRAY[' || tsids || '], ARRAY[' || tbase_ids || '], ARRAY[' || tbase_ids_noagg || ']);')::text is not null then
-        EXECUTE 'update ' || tname || ' set ' || cname || ' = updateid(' || startid || ', gp_segment_id, ARRAY[' || tsids || '], ARRAY[' || tbase_ids || '], ARRAY[' || tbase_ids_noagg || ']);';
-      end if;
-      RETURN '';
-    END;
-    \$\$
-    LANGUAGE 'plpgsql';
-
-    SELECT fast_seqassign('$Table', '$Column', $BeginId);
-" && exit
+    SELECT fast_seqassign('$Table', '$Column', $BeginId, $Increment);
+"
 
+else # if either plpgsql or plpythonu is not available
 # Fall back to using PostgreSQL sequence generator named after the table and column
 # See: http://www.postgresql.org/docs/current/static/sql-createsequence.html
 seq="dd_seq_${Table}_${Column}"
 deepdive sql "
     DROP SEQUENCE IF EXISTS $seq CASCADE;
-    CREATE TEMPORARY SEQUENCE $seq MINVALUE -1 START $BeginId;
+    CREATE TEMPORARY SEQUENCE $seq INCREMENT BY $Increment MINVALUE $(($BeginId - 1)) START $BeginId;
```

Comment:
It's a relic from the Scala code I based my rewrite on.  I wasn't sure if other parts were depending on the particular value, but we can certainly take that out since I'm pretty sure it's unnecessary.